sklearn/tests/metadata_routing_common.py

```diff
@@ -197,14 +201,19 @@ def __init__(self, registry=None, alpha=0.0):
         self.alpha = alpha
         self.registry = registry
 
-    def partial_fit(self, X, y, sample_weight="default", metadata="default"):
+    def partial_fit(
+        self, X, y, classes=None, sample_weight="default", metadata="default"
+    ):
         if self.registry is not None:
             self.registry.append(self)
 
         record_metadata_not_default(
             self, "partial_fit", sample_weight=sample_weight, metadata=metadata
         )
-        self.classes_ = [0, 1]
+        if classes is not None:
+            self.classes_ = classes
+        else:
+            self.classes_ = np.unique(y)
```

Comment:
This does not look like the pattern that we are using classifier. Usually, we will check that `classes_` already exist (is or not the first call to `partial_fit`), if this is the first call then we create `classes_` using `classes` that should not be `None`.

Reply:
yeah but this is a test case, I don't mind having the right thing, but I don't think it matters.

Reply:
@adrinjalali and @glemaitre Please let me know what you prefer and then I can try.

Reply:
After talking with @adrinjalali, I have substitited this part with `_check_partial_fit_first_call(self, classes)`, because this does everything required (check if `classes_` exist, if not using `classes` for it) , also what is done `partial_fit` of `MultiOutputClassifier`, which then had to be fed with initial classes, so that it can pass this check.